packages/product-badge/types.ts

```diff
@@ -0,0 +1,7 @@
+import type { Products } from '@hashicorp/platform-product-meta'
+
+export interface ProductBadgeProps {
+  productName: Products
+  appearance?: 'light' | 'dark'
+  variant?: 'primary' | 'secondary' | 'hasDot'
```

Comment:
💭 primary and secondary feel like a theme, and hasDot feels like a variant.

Thoughts on pulling hasDot into its own prop so the component API might look like:

```tsx
<ProductBadge 
  theme="primary"
/>

// TS error since dot cannot be used in primary theme
<ProductBadge 
  theme="primary"
  dot={true}
/>

<ProductBadge 
  theme="secondary"
/>

<ProductBadge 
  theme="secondary"
  dot={true}
/>
```

Reply:
Love it!